src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -197,29 +197,65 @@ public void OnStartActivity(Activity activity, object payload)
+                        }
+                    }
 
-                    if (request.Host.Port is not null && request.Host.Port != 80 && request.Host.Port != 443)
+                    activity.SetTag(SemanticConventions.AttributeHttpMethod, request.Method);
+                    activity.SetTag(SemanticConventions.AttributeHttpScheme, request.Scheme);
+                    activity.SetTag(SemanticConventions.AttributeHttpTarget, path);
+                    activity.SetTag(SemanticConventions.AttributeHttpUrl, GetUri(request));
+                    activity.SetTag(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocol(request.Protocol));
+
+                    if (request.Headers.TryGetValue("User-Agent", out var values))
                     {
-                        activity.SetTag(SemanticConventions.AttributeNetHostPort, request.Host.Port);
+                        var userAgent = values.Count > 0 ? values[0] : null;
+                        if (!string.IsNullOrEmpty(userAgent))
+                        {
+                            activity.SetTag(SemanticConventions.AttributeHttpUserAgent, userAgent);
+                        }
                     }
                 }
 
-                activity.SetTag(SemanticConventions.AttributeHttpMethod, request.Method);
-                activity.SetTag(SemanticConventions.AttributeHttpScheme, request.Scheme);
-                activity.SetTag(SemanticConventions.AttributeHttpTarget, path);
-                activity.SetTag(SemanticConventions.AttributeHttpUrl, GetUri(request));
-                activity.SetTag(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocol(request.Protocol));
-
-                if (request.Headers.TryGetValue("User-Agent", out var values))
+                // see the spec https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/http.md
+                if (this.httpSemanticConvention.HasFlag(HttpSemanticConvention.New))
                 {
-                    var userAgent = values.Count > 0 ? values[0] : null;
-                    if (!string.IsNullOrEmpty(userAgent))
+                    if (request.Host.HasValue)
+                    {
+                        activity.SetTag(SemanticConventions.AttributeServerAddress, request.Host.Host);
+
+                        if (request.Host.Port is not null && request.Host.Port != 80 && request.Host.Port != 443)
+                        {
+                            activity.SetTag(SemanticConventions.AttributeServerPort, request.Host.Port);
+                        }
+                    }
+
+                    if (request.QueryString.HasValue)
                     {
-                        activity.SetTag(SemanticConventions.AttributeHttpUserAgent, userAgent);
+                        activity.SetTag(SemanticConventions.AttributeUrlQuery, request.QueryString.Value);
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#:~:text=%5B8%5D%3A%20Sensitive%20content%20provided%20in%20query%20string%20SHOULD%20be%20scrubbed%20when%20instrumentations%20can%20identify%20it.

Is there is a way to do this in .NET? need some clarity on this one.

Reply:
That's a good callout. Would need to ask around and see if any other languages are doing this and how.

One point to note, the old code wasn't sanitizing query string: https://github.com/open-telemetry/opentelemetry-dotnet/blob/352f93942a0000743d8c151aca7f05787e33c779/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L432

Reply:
I opened #4571 to track this.

Reply:
Thanks! - A todo comment with link to issue would be helpful